openpype/hosts/houdini/plugins/load/load_redshift_proxy.py

```diff
@@ -0,0 +1,131 @@
+        file_node = container.node("file1")
+        if file_node:
+            file_node.destroy()
+
+        # Add this stub node inside so it previews ok
+        proxy_sop = container.createNode("redshift_proxySOP",
+                                         node_name=node_name)
+        proxy_sop.setDisplayFlag(True)
+
+        nodes = [container, proxy_sop]
+
+        self[:] = nodes
+
+        return pipeline.containerise(
+            node_name,
+            namespace,
+            nodes,
+            context,
+            self.__class__.__name__,
+            suffix="",
+        )
+
+    def update(self, container, representation):
+
+        # Update the file path
+        file_path = get_representation_path(representation)
+
+        node = container["node"]
+        node.setParms({
+            "RS_objprop_proxy_file": self.format_path(file_path)
+        })
+
+        # Update attribute
+        node.setParms({"representation": str(representation["_id"])})
+
+    def remove(self, container):
+
+        node = container["node"]
+        node.destroy()
+
+    def format_path(self, path):
+        """Format using $F{padding} token if sequence, otherwise just path."""
+
+        # Find all frames in the folder
+        ext = ".rs"
+        folder = os.path.dirname(path)
+        frames = [f for f in os.listdir(folder) if f.endswith(ext)]
+
+        # Get the collection of frames to detect frame padding
+        patterns = [clique.PATTERNS["frames"]]
+        collections, remainder = clique.assemble(frames,
+                                                 minimum_items=1,
+                                                 patterns=patterns)
+        self.log.debug("Detected collections: {}".format(collections))
+        self.log.debug("Detected remainder: {}".format(remainder))
+
+        if not collections and remainder:
+            if len(remainder) != 1:
+                raise ValueError("Frames not correctly detected "
+                                 "in: {}".format(remainder))
+
+            # A single frame without frame range detected
+            return os.path.normpath(path).replace("\\", "/")
+
+        # Frames detected with a valid "frame" number pattern
+        # Then we don't want to have any remainder files found
+        assert len(collections) == 1 and not remainder
+        collection = collections[0]
+
+        num_frames = len(collection.indexes)
+        if num_frames == 1:
+            # Return the input path without dynamic $F variable
+            result = path
+        else:
+            # More than a single frame detected - use $F{padding}
+            fname = "{}$F{}{}".format(collection.head,
+                                      collection.padding,
+                                      collection.tail)
+            result = os.path.join(folder, fname)
+
+        # Format file name, Houdini only wants forward slashes
+        return os.path.normpath(result).replace("\\", "/")
```

Comment:
For what it's worth - this is quite legacy code / behavior.
It might be good simplifying this similar to e.g. [this](https://github.com/pypeclub/OpenPype/blob/ba6dfc5eadd1a4d48f92924a2e1bcf3c7d5524f6/openpype/hosts/houdini/plugins/load/load_bgeo.py#L65-L83) (but then of course with `.rs` extension instead of `bgeo`)

By changing that it matches the code style more with that loader, but also the [vdb loader](https://github.com/pypeclub/OpenPype/blob/f58994d59c970878c8d81cb1a1ec19b6748c482a/openpype/hosts/houdini/plugins/load/load_vdb.py#L62-L80) and [ass loader](https://github.com/ynput/OpenPype/blob/c5790fa896f646401e765e2d77527f71e03c7d0a/openpype/hosts/houdini/plugins/load/load_ass.py#L63-L87)

Reply:
> For what it's worth - this is quite legacy code / behavior. It might be good simplifying this similar to e.g. [this](https://github.com/pypeclub/OpenPype/blob/ba6dfc5eadd1a4d48f92924a2e1bcf3c7d5524f6/openpype/hosts/houdini/plugins/load/load_bgeo.py#L65-L83) (but then of course with `.rs` extension instead of `bgeo`)
> 
> By changing that it matches the code style more with that loader, but also the [vdb loader](https://github.com/pypeclub/OpenPype/blob/f58994d59c970878c8d81cb1a1ec19b6748c482a/openpype/hosts/houdini/plugins/load/load_vdb.py#L62-L80) and [ass loader](https://github.com/ynput/OpenPype/blob/c5790fa896f646401e765e2d77527f71e03c7d0a/openpype/hosts/houdini/plugins/load/load_ass.py#L63-L87)

Agreed - the code change is applicable for importing the proxy with frame range